src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,32 @@
+package duke.task;
+
+import duke.exception.EmptyDescriptionException;
+
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    public Task(String description) throws EmptyDescriptionException {
+        if (description.isEmpty()) {
+            throw new EmptyDescriptionException();
+        }
+        this.description = description;
+        this.isDone = false;
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "\u2713" : "\u2718"); // return tick or X symbols
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public String toString() {
+        return String.format("[%s] %s", getStatusIcon(), description);
+    }
+
+    public void markAsDone() {
+        this.isDone = true;
+    }
```

Comment:
Perhaps a good name for your method is setAsDone() as stipulated in the coding standard

Reply:
Changed, thanks.